feat: add password visibility toggle to login form

AbhishekG-Codes opened the pull request:
This PR adds a password visibility toggle to the login form input field, improving usability and accessibility.

- Introduced an eye icon button to toggle between password and plain text visibility.
- Ensured the button is keyboard accessible and screen-reader friendly.
- Used conditional rendering to switch icons based on the visibility state.

This improves user experience, for users who prefer verifying their input before submitting.

No breaking changes introduced.

Closes #129

KarinaOliinyk commented on apps/web/src/app/login/components/login-form.tsx:
```diff
@@ -157,17 +161,28 @@ export function LoginForm({
+                          >
+                            {showPassword ? <EyeOff size={18} /> : <Eye size={18} />}
+                          </button>
+                        </div>
```
@AbhishekG-Codes  It's recommended to review all instances where type="password" is used. Ideally, extract this into a reusable component to ensure consistency and reduce duplication

AbhishekG-Codes replied:
> @AbhishekG-Codes It's recommended to review all instances where type="password" is used. Ideally, extract this into a reusable component to ensure consistency and reduce duplication

Thanks for the input, @KarinaOliinyk! 

We've updated the implementation to follow your recommendation.

Extracted a reusable PasswordInput component to ensure consistency across all input[type="password"] fields.
Replaced all previous password fields across the app with this new component for better maintainability and accessibility.

This ensures the eye icon toggle behavior is standardized and avoids duplication going forward.

Let me know if you’d like any further refinements!

AbhishekG-Codes commented:
Thanks for the input, @KarinaOliinyk!

We've updated the implementation to follow your recommendation.

Extracted a reusable PasswordInput component to ensure consistency across all input[type="password"] fields.
Replaced all previous password fields across the app with this new component for better maintainability and accessibility.

This ensures the eye icon toggle behavior is standardized and avoids duplication going forward.

Let me know if you’d like any further refinements!

0xfurai commented:
@AbhishekG-Codes could you please fix tests fails?
